content/concepts/long-polling.textile

```diff
@@ -0,0 +1,149 @@
+
+h3(#performance-and-scaling). Performance and scaling
+
+Unfortunately, such complexity is difficult to scale effectively. To maintain the session state for a given client, that state must either be sharable among all servers behind a load balancer - a task with significant architectural complexity - or subsequent client requests within the same session must be routed to the same server to which their original request was processed. This form of deterministic "sticky" routing is problematic by design, especially when routing is performed on the basis of IP address, as it can place undue load on a single server in a cluster while leaving other servers mostly idle instead of spreading the load around efficiently. This can also become a potential denial-of-service attack vector - a problem which then requires further layers of infrastructure to mitigate that might otherwise have been unnecessary.
+
+h3(#device-support-and-fallbacks). Device support and fallbacks
+
+In modern times (2018 at the time of this article), long polling can be less relevant for web application development given the widespread availability of real-time communication standards such as WebSockets and WebRTC. That said, there are cases where proxies and routers on certain networks will block WebSocket and WebRTC connections, or where network connectivity can make long-lived connection protocols such as these less practical. Besides, for certain client demographics, there may still be numerous devices and clients in use that lack support for newer standards. For these, long polling can serve as a good fail-safe fallback to ensure support for everyone, irrespective of their situation.
+
+Given that long polling is implemented on the back of @XMLHttpRequest@, which is near universal in its support by devices that still have a non-negligible degree of remaining relevant on the modern web, there's usually not much of a need to support further fallback layers. In cases where exceptions must be handled though, or where a server can be queried for new data but does not support long polling (let alone other more modern technology standards), basic polling can sometimes still be of limited use, and can be implemented using @XMLHttpRequest@, or via JSONP through simple HTML script tags.
+
+Long polling can be implemented via JSONP if absolutely necessary. That said, given the time and effort - not to mention the inefficiency of resource consumption - involved in implementing these approaches, care should be taken to assess whether their support is worth the added cost when developing new applications and system architectures. In many cases, you may be able to get away with exclusive support for something more modern, such as "WebSockets":/concepts/websockets. Alternatively, you may wish to offload the management of these sorts of concerns to a specialist cloud provider such as "Ably":https://ably.io.
```

Comment:
We refer to Ably as "Ably Realtime" in links for SEO reasons.  Also, please use www.ably.io as opposed to ably.io

Reply:
Sure, updated. Thanks